src/overlays/actors/ovl_Bg_Mori_Kaitenkabe/z_bg_mori_kaitenkabe.c

```diff
@@ -20,23 +35,142 @@ const ActorInit Bg_Mori_Kaitenkabe_InitVars = {
+    } else {
+        this->actionFunc = BgMoriKaitenkabe_WaitForMoriTex;
+    }
+}
+
+void BgMoriKaitenkabe_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    s32 pad;
+    BgMoriKaitenkabe* this = THIS;
+    
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgMoriKaitenkabe_WaitForMoriTex(BgMoriKaitenkabe *this, GlobalContext *globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->moriTexObjIndex)) {
+        BgMoriKaitenkabe_SetupWait(this);
+        this->dyna.actor.draw = BgMoriKaitenkabe_Draw;
+    }
+}
+
+void BgMoriKaitenkabe_SetupWait(BgMoriKaitenkabe* this) {
+    this->actionFunc = BgMoriKaitenkabe_Wait;
+    this->timer = 0;
+}
+
+void BgMoriKaitenkabe_Wait(BgMoriKaitenkabe *this, GlobalContext *globalCtx) {
+    Vec3f sp3C;
+    Vec3f sp30;
+    Vec3f sp24;
+    Player* player = PLAYER;
+
+    if (0.001f < this->dyna.unk_150) {
+        this->timer++;
+        if ((this->timer > 28) && !Player_InCsMode(globalCtx)) {
+            BgMoriKaitenkabe_SetupRotate(this);
+            func_8002DF54(globalCtx, &this->dyna.actor, 8);
+            Math_Vec3f_Copy(&this->playerPos, &player->actor.posRot.pos);
+            sp3C.x = Math_Sins(this->dyna.unk_158);
+            sp3C.y = 0.0f;
+            sp3C.z = Math_Coss(this->dyna.unk_158);
+            sp30.x = this->dyna.actor.posRot.pos.x - player->actor.posRot.pos.x;
+            sp30.y = 0.0f;
+            sp30.z = this->dyna.actor.posRot.pos.z - player->actor.posRot.pos.z;
+            BgMoriKaitenkabe_CrossProd(&sp24, &sp3C, &sp30);
+            this->rotDirection = (sp24.y > 0.0f) ? 1.0f : -1.0f;
+        }
+    } else {
+        this->timer = 0;
+    }
+    if (fabsf(this->dyna.unk_150) > 0.001f) {
+        this->dyna.unk_150 = 0.0f;
+        player->stateFlags2 &= ~0x10;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Kaitenkabe/BgMoriKaitenkabe_Destroy.s")
+void BgMoriKaitenkabe_SetupRotate(BgMoriKaitenkabe *this) {
+    this->actionFunc = BgMoriKaitenkabe_Rotate;
+    this->rotSpeed = 0.0f;
+    this->rotDist = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Kaitenkabe/func_808A4F3C.s")
+void BgMoriKaitenkabe_Rotate(BgMoriKaitenkabe *this, GlobalContext *globalCtx) {
+    Player* player = PLAYER;
+    s16 tempRot;
+    
+    Math_ApproxF(&this->rotSpeed, 0.6f, 0.02f);
+    if (Math_ApproxF(&this->rotDist, this->rotDirection * 45.0f, this->rotSpeed)) {
+        BgMoriKaitenkabe_SetupWait(this);
+        func_8002DF54(globalCtx, &this->dyna.actor, 7);
+        if (this->rotDirection > 0.0f) {
+            this->dyna.actor.initPosRot.rot.y += 0x2000;
+        } else {
+            this->dyna.actor.initPosRot.rot.y -= 0x2000;
+        }
+        this->dyna.actor.posRot.rot.y = this->dyna.actor.shape.rot.y = this->dyna.actor.initPosRot.rot.y;
+        func_800788CC(NA_SE_EV_STONEDOOR_STOP);
+    } else {
+        tempRot = this->rotDist * (0x10000/360.0f);
```

Comment:
```suggestion
        rotY = this->rotYdeg * (0x10000/360.0f);
```
I guess the Y can be dropped since this actor is mostly known for rotating and it should be pretty obvious, avoids putting it everywhere else